homeassistant/components/zwave_js/services.py

```diff
@@ -0,0 +1,175 @@
+from zwave_js_server.util.node import async_set_config_parameter
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_DEVICE_ID, ATTR_ENTITY_ID
+from homeassistant.core import HomeAssistant, ServiceCall, callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.device_registry import DeviceRegistry
+from homeassistant.helpers.entity_registry import EntityRegistry
+
+from . import const
+from .entity import get_home_and_node_id_from_device_id
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def parameter_name_does_not_need_bitmask(
+    val: Dict[str, Union[int, str]]
+) -> Dict[str, Union[int, str]]:
+    """Validate that if a parameter name is provided, bitmask is not as well."""
+    if isinstance(val[const.ATTR_CONFIG_PARAMETER], str) and (
+        val.get(const.ATTR_CONFIG_PARAMETER_BITMASK)
+    ):
+        raise vol.Invalid(
+            "Don't include a bitmask when a parameter name is specified",
+            path=[const.ATTR_CONFIG_PARAMETER, const.ATTR_CONFIG_PARAMETER_BITMASK],
+        )
+    return val
+
+
+# Validates that a bitmask is provided in hex form and converts it to decimal
+# int equivalent since that's what the library uses
+BITMASK_SCHEMA = vol.All(
+    cv.string, vol.Lower, vol.Match(r"^(0x)?[0-9a-f]+$"), lambda value: int(value, 16)
+)
+
+
+class ZWaveServices:
+    """Class that holds our services (Zwave Commands) that should be published to hass."""
+
+    def __init__(
+        self, hass: HomeAssistant, dev_reg: DeviceRegistry, ent_reg: EntityRegistry
+    ):
+        """Initialize with hass, device registry, and entity registry objects."""
+        self._hass = hass
+        self._dev_reg = dev_reg
+        self._ent_reg = ent_reg
+
+    @callback
+    def async_register(self) -> None:
+        """Register all our services."""
+        self._hass.services.async_register(
+            const.DOMAIN,
+            const.SERVICE_SET_CONFIG_PARAMETER,
+            self.async_set_config_parameter,
+            schema=vol.All(
+                {
+                    vol.Exclusive(ATTR_DEVICE_ID, "id"): cv.string,
+                    vol.Exclusive(ATTR_ENTITY_ID, "id"): cv.entity_ids,
+                    vol.Required(const.ATTR_CONFIG_PARAMETER): vol.Any(
+                        vol.Coerce(int), cv.string
+                    ),
+                    vol.Optional(const.ATTR_CONFIG_PARAMETER_BITMASK): vol.Any(
+                        vol.Coerce(int), BITMASK_SCHEMA
+                    ),
+                    vol.Required(const.ATTR_CONFIG_VALUE): vol.Any(
+                        vol.Coerce(int), cv.string
+                    ),
+                },
+                cv.has_at_least_one_key(ATTR_DEVICE_ID, ATTR_ENTITY_ID),
+                parameter_name_does_not_need_bitmask,
+            ),
+        )
+
+    @callback
+    def async_get_node_from_device_id(self, device_id: str) -> ZwaveNode:
```

Comment:
maybe use this to entity.py or even util/helper file ?
We also need this for device triggers